Twitch EventSub library/EventSubSocketWrapper.cs

```diff
@@ -432,15 +437,28 @@ private async Task ReconnectMessageProcessingAsync(WebSocketReconnectMessage mes
 
             _awaitForReconnect = true;
             _watchdog.Stop();
-            await _socket.DisconnectAsync();
+
             if (message?.Payload?.Session.ReconnectUrl != null)
             {
-                _connectionActive = await _socket.ConnectAsync(message.Payload.Session.ReconnectUrl);
-                if (!_connectionActive)
+                if (Uri.TryCreate(message.Payload.Session.ReconnectUrl, new UriCreationOptions() { DangerousDisablePathAndQueryCanonicalization = false }, out var Url))
+                {
+                    _socket.Url = Url;
+                    await _socket.ReconnectOrFail();
+                    _socket.MessageReceived.Select(msg => Observable.FromAsync(async () => await SocketOnMessageReceivedAsync(this, msg.Text))).Concat().Subscribe();
```

Comment:
I don't think you should be subscribing again. I didn't read the docs of the implementation, but I would say t hat you register the events only once (during socket setup). This is just URL change. Or do you have it verified it needs to be done after URL change?

Reply:
Reconnect throws away entire object. This is from documentation.